File: Exercism Codes/Java/Flattener.java
import java.util.ArrayList;
import java.util.List;

class Flattener {
    public List<Object> flatten(List<Object> list){
        List<Object> result = new ArrayList<Object>();
        for (Object element : list) {
            if(element == null) continue;
            if (element instanceof List) {
                result.addAll(flatten((List<Object>) element));
            } else {
                result.add(element);
            }
        }
        return result;
    }
}

File: Exercism Codes/Kotlin/Flattener.kt
object Flattener {
    fun flatten(source: Collection<Any?>): List<Any> {
        val result = mutableListOf<Any>()
        for (element in source) {
            when (element) {
                null -> 0
                is Collection<*> -> result.addAll(flatten(element))
                else -> result.add(element)
            }
        }
        return result
    }
}
